app/code/Magento/ConfigurableProduct/Model/ResourceModel/Product/Type/Configurable/StockStatus.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ConfigurableProduct\Model\ResourceModel\Product\Type\Configurable;
+
+use Magento\Framework\EntityManager\MetadataPool;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\CatalogInventory\Api\Data\StockStatusInterface as CatalogInventoryStockStatusInterface;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+
+class StockStatus implements StockStatusInterface
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * @var MetadataPool
+     */
+    private $metadataPool;
+
+    /**
+     * StockStatus constructor.
+     * @param ResourceConnection $resource
+     * @param MetadataPool $metadataPool
+     */
+    public function __construct(ResourceConnection $resource, MetadataPool $metadataPool)
+    {
+        $this->resource = $resource;
+        $this->metadataPool = $metadataPool;
+    }
+
+    /**
+     * @var array
+     */
+    private $allChildOutOfStockInfo = [];
+
+    /**
+     * @inheritdoc
+     */
+    public function isAllChildOutOfStock(int $productId): bool
+    {
+        if (isset($this->allChildOutOfStockInfo[$productId])) {
+            return $this->allChildOutOfStockInfo[$productId];
+        }
+
+        $statuses = $this->getAllChildStockInfo($productId);
+        $isAllChildOutOfStock = true;
+        foreach ($statuses as $status) {
+            if ($status === CatalogInventoryStockStatusInterface::STATUS_IN_STOCK) {
+                $isAllChildOutOfStock = false;
+                break;
+            }
+        }
+
+        $this->allChildOutOfStockInfo[$productId] = $isAllChildOutOfStock;
+        return $this->allChildOutOfStockInfo[$productId];
```

Comment:
Please a blank line before return statement

Reply:
done

Reply:
Done

Reply:
@novikor not needed, please concentrate on crucial things during review.